crates/red_knot/src/types/infer.rs

```diff
@@ -36,15 +34,27 @@ where
             // TODO relative imports
             assert!(matches!(level, 0));
             let module_name = ModuleName::new(module.as_ref().expect("TODO relative imports"));
-            if let Some(module) = db.resolve_module(module_name) {
+            if let Some(module) = db.resolve_module(module_name.clone()) {
                 let remote_file_id = module.path(db).file();
                 let remote_symbols = db.symbol_table(remote_file_id);
                 if let Some(remote_symbol_id) = remote_symbols.root_symbol_id_by_name(name) {
+                    // TODO integrate this into module and symbol-resolution APIs (requiring a
+                    // "requester" argument) so that it doesn't have to be remembered
+                    db.jar().type_store.record_symbol_dependency(
+                        (file_id, symbol_id),
```

Comment:
I think I already commented this on the other PR. I think it would be nice if we had a struct that combines `file_id` and `symbol_id`. I find that clearer than unnamed two-element tuples. We may then even change the `infer_symbol_type` method to take such an instance as the argument instead of a file id and symbol id pair.

Reply:
Yep, agreed, will push that new type in this PR.